Add unit tests to cover DecoderNLS.Convert 'completed' scenarios

GrabYourPitchforks opened the pull request:
Unit tests for https://github.com/dotnet/coreclr/pull/27210. These unit tests will not pass until the nightly coreclr build makes its way to corefx.

stephentoub commented on src/System.Text.Encoding/tests/Decoder/DecoderConvert2.cs:
```diff
@@ -155,6 +155,7 @@ public void PosTest5()
 
         // PosTest6: Call Convert to convert a ASCII character array with Unicode encoder
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "coreclr #27191 is not fixed in netfx.")]
```
We don't appear to run these tests on netfx:
https://github.com/dotnet/corefx/blob/90b15f766dc78bcb9246eff6b64c02220dacb08f/src/System.Text.Encoding/tests/Configurations.props#L4-L5
so I wouldn't think this attribution would be necessary.

stephentoub commented on src/System.Text.Encoding/tests/Decoder/DecoderConvert2.cs:
```diff
@@ -274,6 +275,30 @@ public void PosTest10()
 
+        // PosTest11: Call Convert with UTF-8 data, testing that flushing performs proper replacement if needed
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "coreclr #27191 is not fixed in netfx.")]
```
Ditto

GrabYourPitchforks commented:
This is probably going in with https://github.com/dotnet/corefx/pull/41851. I've updated that PR to match this one and removed the unneeded attributes per Steve's recommendations.
